libcxx/include/__type_traits/is_trivially_relocatable.h

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___TYPE_TRAITS_IS_TRIVIALLY_RELOCATABLE_H
+#define _LIBCPP___TYPE_TRAITS_IS_TRIVIALLY_RELOCATABLE_H
+
+#include <__config>
+#include <__type_traits/is_trivially_destructible.h>
+#include <__type_traits/is_trivially_move_constructible.h>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+//------------------------------------------------------WARNING-------------------------------------------------------//
+//
+// This type trait should only be used where ABI stability isn't relevant or can be assured by other means,
+// like in function bodies or where the difference can be detected at runtime. If this trait is used in ABI-sensitive
+// environments it has to be ensured that Clang and GCC can interpret the data produced by each other. If you are not
+// certain that this is the case DO NOT USE THIS TYPE TRAIT
+//
+//--------------------------------------------------------------------------------------------------------------------//
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+#if __has_builtin(__is_trivially_relocatable)
```

Comment:
I wasn't aware. Thanks for the info. Maybe I'll try to fix it. I haven't had a lot of luck with clang patches yet. Probably because I lost interest in the one where I actually planned to land it at some point. Maybe you could help me a bit if I make a patch?

Reply:
Sure!